### PR TITLE
Content gets attributes

### PR DESCRIPTION
**Pull request by cstamas**
Makes `MavenContent` gone, and opens possibility for proxy facet to give extra information (like would be origin URL, TBD) of the content it got and needs to be cached.

### CODE REVIEW

File: plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenHashCodes.java [hunk 1]
```diff
@@ -19,44 +19,35 @@
...
 
 /**
- * Maven content.
+ * Maven hash codes map, usable as attribute.
  *
  * @since 3.0
  */
-public class MavenContent
-    extends Content
+public class MavenHashCodes
```
Comment (adreghiciu) [inline plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenHashCodes.java]: What is the advantage of having this class vs. directly storing the hashes map?

Comment (jdillon) [inline plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenHashCodes.java]: Agree this seems like a pointless wrapper.  Also hash-codes are not really specific to maven.  So perhaps consider an attribute that attaches these generally?

Comment (cstamas) [inline plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenHashCodes.java]: Ack, I agree. @adreghiciu removing wrapper and moving them to more general place.

**Comment by cstamas** [comment]
Superseded by https://github.com/sonatype/nexus-oss/pull/1146

**Pull request closed without merging**